virtualizarr/readers/hdf/hdf.py

```diff
@@ -88,6 +81,22 @@ def _construct_manifest_array(
         attrs = HDFVirtualBackend._extract_attrs(dataset)
         dtype = dataset.dtype
 
+        # Temporarily disable use CF->Codecs - TODO re-enable in subsequent PR.
+        # cfcodec = cfcodec_from_dataset(dataset)
+        # if cfcodec:
+        # codecs.insert(0, cfcodec["codec"])
+        # dtype = cfcodec["target_dtype"]
+        # attrs.pop("scale_factor", None)
+        # attrs.pop("add_offset", None)
+        # else:
+        # dtype = dataset.dtype
```

Comment:
What are the implications of disabling this? Will virtualization not work properly for datasets with a CF-style scale and offset?

Reply:
@maxrjones When I first created the reader I used this as a proof of concept to demonstrate using codecs for handling CF encoding to make data decoding more accessible for non-xarray clients https://github.com/zarr-developers/VirtualiZarr/issues/68#issuecomment-2048034811.  This works as expected but now causes failure of existing roundtrip tests now that we're not directly creating xarray structures.  To reduce the scope of the PR I've disabled it here (but the CF attrs are still carried through so all tests pass).  I'll re-enable this in a separate PR.